src/app/header/header.component.ts

```diff
@@ -0,0 +1,30 @@
+import { Component, OnDestroy, OnInit } from "@angular/core";
+import { Subscriber, Subscription } from "rxjs";
+import { AuthService } from "../auth/auth.service";
+
+@Component({
+  selector: "app-header",
+  templateUrl: "./header.component.html"
+})
+export class HeaderComponent implements OnInit, OnDestroy{
+
+  private userSubsciption: Subscription = new Subscription();
+  isAuthenticated = false;
+
+  constructor(private authService:AuthService){}
+
+  ngOnInit(): void {
+    this.authService.user.subscribe(user=>{
+      this.isAuthenticated = !!user;
+    })
+    this.userSubsciption.add(this.userSubsciption);
```

Comment:
this line is totally wrong, you add userSubscription to userSubscription

it should be 

```
const subsription = this.authService.....

this.userSubsciption.add(subsription);

```